ARACHNE-2768 Datanode 1.15 does not use proxy settings in case of ARA…

YaroslavTir opened the pull request:
Arachne 2768 add proxy supprot for rest authentication

Commit:
ARACHNE-2768 Datanode 1.15 does not use proxy settings in case of ARACHNE central authentication (Regenstrief production issue)
- add proxy support for rest authentication.

pavgra commented on src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java:
```diff
@@ -0,0 +1,58 @@
+
+    public static final String TEST_URL = "https://www.arachnenetwork.com/api/v1/build-number";
+    //this proxy ip is not fixed, so it could change. To fix this use any available proxy
+    public static final String PROXY_URL = "167.99.232.18";
```
Where is this magic IP from? Cannot we avoid having new tests requiring an internet connection to complete?

YaroslavTir replied:
good point, I don't like this either. This is a random proxy that I have found on the internet. I plan to replace this one with our own proxy that Konstantin is working on. There is an option to set up an embedded proxy for all unit tests(timewise, I am not sure that it worth it ). and the last option is just to get rid of the test.

pavgra replied:
> I plan to replace this one with our own proxy that Konstantin is working on

That doesn't solve the conceptual issue

> There is an option to set up an embedded proxy for all unit tests

Yes, that's what we should go for

YaroslavTir replied:
fixed

Commit:
ARACHNE-2768 Datanode 1.15 does not use proxy settings in case of ARACHNE central authentication (Regenstrief production issue)
- add embedded proxy for unit tests

Commit:
ARACHNE-2768 Datanode 1.15 does not use proxy settings in case of ARACHNE central authentication (Regenstrief production issue)
- do not remove headers from client request in TestProxy. This step is not necessary.

pavgra commented on src/test/java/org/ohdsi/authenticator/service/rest/RestTemplateProviderTest.java:
```diff
@@ -0,0 +1,66 @@
+        RestTemplate restTemplate = restTemplateProvider.createRestTemplate();
+        ResponseEntity<String> response = restTemplate.exchange(TEST_URL, HttpMethod.GET, null, String.class);
+
+        assertEquals(HttpStatus.OK, response.getStatusCode());
```
I would add Mockito and check that proxy has actually been invoked

YaroslavTir replied:
fixed

pavgra commented on src/test/java/org/ohdsi/authenticator/service/proxy/ProxyInitializer.java:
```diff
@@ -0,0 +1,29 @@
+
+
+/**
+ * Proxy should be start only ONCE for all unit tests!!
```
Language

YaroslavTir replied:
fixed

Commit:
ARACHNE-2768 Datanode 1.15 does not use proxy settings in case of ARACHNE central authentication (Regenstrief production issue)
- ProxyUnit test: check that proxy was invoked.

Commit:
ARACHNE-2768 Datanode 1.15 does not use proxy settings in case of ARACHNE central authentication (Regenstrief production issue)
- ProxyUnit test: small clean up
